tool/plugin/lib/plugin.dart

```diff
@@ -624,6 +627,48 @@ class GenerateCommand extends ProductCommand {
 
   SyntheticBuildSpec makeSyntheticSpec(List specs) =>
       SyntheticBuildSpec.fromJson(specs[0], release, specs[2]);
+
+  void generateLiveTemplates() {
+    // Find all the live templates.
+    final templateFragments = Directory(p.join('resources', 'liveTemplates'))
+        .listSync()
+        .where((entity) => entity is File)
+        .where((file) => p.extension(file.path) == 'txt')
+        .cast<File>()
+        .toList();
+    final templateFile =
+        File(p.join('resources', 'liveTemplates', 'flutter_miscellaneous.xml'));
+    var contents = templateFile.readAsStringSync();
+
+    log('writing ${p.relative(templateFile.path)}');
+
+    for (var file in templateFragments) {
+      final name = p.basenameWithoutExtension(file.path);
+
+      var replaceContents = file.readAsStringSync();
+      replaceContents = replaceContents
+          .replaceAll('\n', '&#10;')
+          .replaceAll('<', '&lt;')
+          .replaceAll('>', '&gt;');
+
+      // look for '<template name="$name" value="..."'
+      final regexp = RegExp('<template name="$name" value="([^"]+)"');
```

Comment:
Maybe add a README explaining double quotes are not allowed and new templates have to be manually added to the xml file?

Reply:
Good idea - I'll add one.